### ISSUE TITLE
Loss no convergence

### ISSUE THREAD

**Issue by TengliEd** [syfafterzy/SVDNet-for-Pedestrian-Retrieval#9]
I downloaded the raw caffemodel from http://dl.caffe.berkeleyvision.org/bvlc_reference_caffenet.caffemodel and modified the data paths and then trained it using the provided config prototxt file (solver_linear.prototxt). But loss did not converge around 6.5. What else modifications are needed?

**Comment by TengliEd** [comment]
Or someone who has repeated this experiment can specify steps to make it from this code repository? Cheers!

**Comment by ghost** [comment]
Maybe you can show your log, and you can also check the IDE baseline (https://github.com/zhunzhong07/IDE-baseline-Market-1501).

发送自 Windows 10 版邮件应用

发件人: Teng Li
发送时间: 2018年3月13日 12:00
收件人: syfafterzy/SVDNet-for-Pedestrian-Retrieval
抄送: Subscribed
主题: Re: [syfafterzy/SVDNet-for-Pedestrian-Retrieval] Loss no convergence(#9)

Or someone who has repeated this experiment can specify steps to make it from this code repository? Cheers!

**Comment by TengliEd** [comment]
[train_linear.log](https://github.com/syfafterzy/SVDNet-for-Pedestrian-Retrieval/files/1805972/train_linear.log)

**Comment by TengliEd** [comment]
I suppose I made bad lmdb. Could you plz provide files to produce lmdb data?

**Comment by ghost** [comment]
1. I check you log, and Finetuning from SVDNet/caffenet/1024d_linear.caffemodel seems strange.

Do you train 1024d_linear.caffemodel by yourself?
Step 2: Train a model with a linear fully-connected layer based on caffenet or resnet.

2. About LMDB:

 see here https://github.com/syfafterzy/SVDNet-for-Pedestrian-Retrieval/issues/1
or http://caffe.berkeleyvision.org/gathered/examples/imagenet.html

**Comment by TengliEd** [comment]
Many Thanks @Simon4john . I found I used person IDs  as labels. Now I have no idea extracting features on test data to evaluate.

**Comment by ghost** [comment]
@TengliEd 
To evaluate the trained model, you can see [here](https://github.com/zhunzhong07/IDE-baseline-Market-1501)

Good luck!

**Comment by TengliEd** [comment]
I evaluated the caffenet trained by RRI but results did not achieve as high level as mentioned in the paper (R1: 72.62%; mAP: 44.85%). I used Euclidean distance to measure similarity. Should I use other measurements to achieve your model performance?

**Comment by ghost** [comment]
During testing, the image needs to subtract mean. Be sure the mean is same for training and testing.

**Comment by ghost** [comment]
测试时候，需要减去的均值文件和训练的必须保持一致，你检查一下。

**Comment by wjzh1** [comment]
@Simon4john 请问我用的是resnet18网络，在分类层前面加了一个fc层，在用作者matlab代码转换的时候出现索引维度超出范围，请问要怎么修改

**Comment by wjzh1** [comment]
@Simon4john 请问我用的是resnet18网络，在分类层前面加了一个fc层，在用作者matlab代码转换的时候出现索引超出矩阵维度，请问要怎么修改change_resnet_W.m这代码

**Comment by ghost** [comment]
@wjzh1 检查一下https://github.com/syfafterzy/SVDNet-for-Pedestrian-Retrieval/blob/439ae0e086304cc8e38c8112b2a96e1a8ebdf647/matlab/SVDNet/change_resnet_W.m#L1
deploy renset的时候你得prototext和model是对应的吗？按照给了resnet50的案例照着理一遍，就可以训练出来的。仔细一点。

**Comment by wjzh1** [comment]
------------------ 原始邮件 ------------------
发件人: "John"<notifications@github.com>;
发送时间: 2018年4月16日(星期一) 下午4:58
收件人: "syfafterzy/SVDNet-for-Pedestrian-Retrieval"<SVDNet-for-Pedestrian-Retrieval@noreply.github.com>;
抄送: "beyond"<1678824369@qq.com>; "Mention"<mention@noreply.github.com>; 
主题: Re: [syfafterzy/SVDNet-for-Pedestrian-Retrieval] Loss no convergence(#9)




@wjzh1 检查一下https://github.com/syfafterzy/SVDNet-for-Pedestrian-Retrieval/blob/439ae0e086304cc8e38c8112b2a96e1a8ebdf647/matlab/SVDNet/change_resnet_W.m#L1
 deploy renset的时候你得prototext和model是对应的吗？按照给了resnet50的案例照着理一遍，就可以训练出来的。仔细一点。
